redisinsight/api/src/modules/cloud/subscription/cloud-subscription.api.service.ts

```diff
@@ -0,0 +1,85 @@
+import { filter, find } from 'lodash';
+import { Injectable, Logger } from '@nestjs/common';
+import { SessionMetadata } from 'src/common/models';
+import { wrapHttpError } from 'src/common/utils';
+import { CloudCapiAuthDto } from 'src/modules/cloud/common/dto';
+import { CloudRequestUtm, ICloudApiCredentials } from 'src/modules/cloud/common/models';
+import { CloudSubscriptionPlanResponse } from './models/api.interface';
+import { CloudSubscriptionCapiService } from './cloud-subscription.capi.service';
+import { CloudUserApiService } from '../user/cloud-user.api.service';
+import { CloudSubscriptionRegion, CloudSubscriptionType } from './models';
+import { CloudSessionService } from '../session/cloud-session.service';
+import { parseCloudSubscriptionsCloudRegionsCapiResponse } from './utils';
+import { CloudSession } from '../session/models/cloud-session';
+import { CloudSubscriptionApiProvider } from './providers/cloud-subscription.api.provider';
+
+@Injectable()
+export class CloudSubscriptionApiService {
+  private logger = new Logger('CloudSubscriptionApiService');
+
+  constructor(
+    private readonly api: CloudSubscriptionApiProvider,
+    private readonly sessionService: CloudSessionService,
+    private readonly cloudUserApiService: CloudUserApiService,
+    private readonly cloudSubscriptionCapiService: CloudSubscriptionCapiService,
+  ) {}
+
+  private async getApiCredentials(sessionId: string): Promise<CloudSession> {
+    return this.sessionService.getSession(sessionId);
+  }
+
+  private async getCapiCredentials(sessionMetadata: SessionMetadata, utm?: CloudRequestUtm): Promise<CloudCapiAuthDto> {
```

Comment:
and for this.

Reply:
for the future methods

Reply:
I don't want to add these methods to all services. We are going to repeat ourselves then
getCapiCredentials and  getApiCredentials should be moved to corresponding services then
anyway you can do the same as for other services for now